src/IO/GeoJSONWriter.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Brick\Geo\IO;
+
+use Brick\Geo\Exception\GeometryIOException;
+use Brick\Geo\Geometry;
+use Brick\Geo\GeometryCollection;
+use Brick\Geo\LineString;
+use Brick\Geo\MultiLineString;
+use Brick\Geo\MultiPoint;
+use Brick\Geo\MultiPolygon;
+use Brick\Geo\Point;
+use Brick\Geo\Polygon;
+
+/**
+ * Converter class from Geometry to GeoJSON.
+ */
+class GeoJSONWriter
+{
+    /**
+     * @param Geometry $geometry The geometry to export as GeoJSON.
+     *
+     * @return string The GeoJSON representation of the given geometry.
+     *
+     * @throws GeometryIOException If the given geometry cannot be exported as GeoJSON.
+     */
+    public function write(Geometry $geometry) : string
+    {
+        if ($geometry instanceof GeometryCollection
+            && ! is_subclass_of($geometry, GeometryCollection::class)) {
+            return $this->writeFeatureCollection($geometry);
+        }
+
+        return $this->genGeoJSONString($this->formatGeoJSONGeometry($geometry));
+    }
+
+    /**
+     * @param Geometry $geometry
+     *
+     * @return array
+     *
+     * @throws GeometryIOException
+     */
+    private function formatGeoJSONGeometry(Geometry $geometry) : array
+    {
+        if ($geometry instanceof Point) {
+            $type = 'Point';
+        } elseif ($geometry instanceof MultiPoint) {
+            $type = 'MultiPoint';
+        } elseif ($geometry instanceof LineString) {
+            $type = 'LineString';
+        } elseif ($geometry instanceof MultiLineString) {
+            $type = 'MultiLineString';
+        } elseif ($geometry instanceof Polygon) {
+            $type = 'Polygon';
+        } elseif ($geometry instanceof MultiPolygon) {
+            $type = 'MultiPolygon';
+        } else {
+            throw GeometryIOException::unsupportedGeometryType($geometry->geometryType());
+        }
```

Comment:
If think this whole code block can be greatly reduced, as all Geometries provide a `geometryType()` method that returns the exact same string (CamelCased as well), which you could use?

Maybe you should only check that this string is one of the 6 allowed, and then use it for `$type`.

Reply:
That's exactly what I had in mind: `'type' => $geometry->geometryType()`.

Regarding the checks however, I would favour the `in_array()` approach:

```
if (! in_array($geometryType, ['Point', 'MultiPoint', ...])) {
    throw ...
}
```

Otherwise the benefit is limited. Note that the array of supported GeoJSON types could be in a class constant to keep the code clean & readable.

I get your point regarding the `instanceof` and the possibility to look up class usages, but I'm not sure how useful it is here.

Reply:
Resolved in Commit: 151006b